qiskit/providers/fake_provider/fake_mumbai_v2.py

```diff
@@ -30,11 +30,19 @@
 from qiskit.providers.backend import BackendV2, QubitProperties
 from qiskit.providers.options import Options
 from qiskit.transpiler import Target, InstructionProperties
+from qiskit.utils.deprecation import deprecate_func
 
 
 class FakeMumbaiFractionalCX(BackendV2):
     """A fake mumbai backend."""
 
+    @deprecate_func(
+        additional_msg="Use the `qiskit.providers.basic_provider.GenericBackendV2` "
+        "class instead.",
+        since="0.46.0",
+        removal_timeline="Qiskit 1.0",
+        package_name="qiskit",
+    )
     def __init__(self):
         super().__init__(
             name="FakeMumbaiFractionalCX",
```

Comment:
From the [other PR](https://github.com/Qiskit/qiskit/pull/10952), I see in the description:

> The FakeMumbaiFractionalCX is a mumbai-inspired test-specific backend and is not part of the deprecation/migration to provider plan. I think that we could either leave it as-is or rename it to something more general like "FakeFractionalCX".

Does this mean we shouldn't deprecate this class here?

Reply:
I had a brief chat with Matthew this week and we agreed to deprecate it and move the class out of the public API (to `test`), as this is a very specific backend to test a very specific implementation. The only non-deprecated classes would be the ones part of the "V1 hierarchy":
- V1 base classes + the "generic V1 alternatives"
- `Fake1Q` (it's one of the "generic V1 alternatives")
- `FakeOpenpulse2q` and `FakeOpenpulse3q `(they are actively maintained and used on pulse tests, also V1)